server/controllers/profile.js

```diff
@@ -0,0 +1,82 @@
+const Profile = require("../models/Profile");
+const asyncHandler = require("express-async-handler");
+
+exports.profileCreatePost = asyncHandler(async (req, res) => {
+  const {
+    firstName,
+    lastName,
+    description,
+    availability,
+    email,
+    username,
+    gender,
+    birthday,
+    phone,
+    address,
+    available,
+  } = req.body;
+  const id = req.user.id;
+
+  if (!firstName || !lastName || !id) {
+    res.status(400);
+    throw new Error("Invalid name");
+  }
+  const profile = await Profile.create({
+    userId: id,
+    firstName,
+    lastName,
+    description,
+    availability,
+    email,
+    username,
+    gender,
+    birthday,
+    phone,
+    address,
+    available,
+  });
+  res.status(201).json({ profile });
+});
+
+exports.profileUpdatePost = asyncHandler(async (req, res) => {
+  const { firstName, lastName, description, availability, email } = req.body;
+  const id = req.user.id;
+  const idExists = await Profile.findOne({ _id: id });
+
+  if (idExists) {
+    const update = await Profile.updateOneById(
+      { id },
+      {
+        username,
+        firstName,
+        email,
+        lastName,
+        description,
+        availability,
+      }
+    );
+    res.status(200).json({ update: update });
+  } else {
+    res.status(500);
+    throw new Error("Invalid profile update");
+  }
+});
+
+exports.profileGet = asyncHandler(async (req, res) => {
+  const id = req.params.id ? req.params.id : req.user.id;
```

Comment:
```suggestion
  const userId = req.user.id;
```

This route does not have route parameters defined :)